03-Raster-Calculations-In-R.Rmd

```diff
@@ -143,97 +172,78 @@ CHM_HARV <- DSM_HARV - DTM_HARV
 hist(CHM_HARV,
      col="springgreen4",
      main="Histogram of NEON Canopy Height Model\nHarvard Forest",
-     ylab="Tree Height (m)",
-     xlab="Number of Pixels")
+     ylab="Number of Pixels",
+     xlab="Tree Height (m) ")
 
 ```
 
-
-Notice that the range of values for the output `CHM` is between 0 and 30 meters.
+Notice that the range of values for the output CHM is between 0 and 30 meters.
 Does this make sense for trees in Harvard Forest?
 
-
-****
 #Challenge: Exploring Raster Values in CHM
-As we learned in previous lessons, it's often a good idea to explore the range 
-of values in a raster dataset just like we might explore a dataset that we 
-collected in the field.  
+It's often a good idea to explore the range of values in a raster dataset just
+like we might explore a dataset that we collected in the field.  
 
 1. What is the min and maximum value for the Harvard Forest Canopy 
 Height Model (`CHM_HARV`) that we just created?
 2. What are two ways you can check this range of data in `CHM_HARV`? 
-3. What is the distribution of the pixel values in the CHM? Plot a histogram with 
-6 bins instead of the default.
-4. Bonus challenge: Change the color of the histogram you created 
-in question 3. 
-5. Plot the `CHM_HARV` raster using the breaks that make sense for the data. You 
-mayb consider 3-4 breakpoints. Generate a set of colors that suites the data. Add
-a title to the plot and turn off the axes.
-****  
-  
-Next, we will perform the same calculations using a more efficient aproach - the
-`overlay()` function.  
-
-```{r challenge-code-CHM-HARV, echo=FALSE }
-#1) Depends on types of trees.  0m to <50m?
-#2) Looks at histogram or min/max values. 
+3. What is the distribution of all the pixel values in the CHM? 
+4. Plot a histogram with 6 bins instead of the default and change the color of
+the histogram. 
+5. Plot the `CHM_HARV` raster using breaks that make sense for the data. Include
+a appropriate color palette for the data, plot title and no axes. 
+
+```{r challenge-code-CHM-HARV,  include=TRUE, results="hide", echo=FALSE} 
+#1) 
 minValue(CHM_HARV)
 maxValue(CHM_HARV)
-#3) 
+#2) Looks at histogram, minValue(NAME)/maxValue(NAME), NAME and look at values slot. 
+#3
 hist(CHM_HARV, 
-     maxpixels=ncell(CHM_HARV),
      col="springgreen4",
-     main = "Histogram of NEON Canopy Height Model\n Harvard Forest",)
-
-#4)
+     main = "Histogram of NEON Canopy Height Model\n Harvard Forest",
+     maxpixels=ncell(CHM_HARV))
+#4 
 hist(CHM_HARV, 
-     col = "springgreen4",
+     col="lightgreen",
      main = "Histogram of NEON Canopy Height Model\n Harvard Forest",
-     maxpixels=ncell(CHM),
+     maxpixels=ncell(CHM_HARV),
      breaks=6)
-
-
-
-```
-
-```{r challenge-plot, echo=FALSE }
-
-#5)
+#5
 myCol=terrain.colors(4)
 plot(CHM_HARV,
      breaks=c(0,10,20,30),
      col=myCol,
      axes=F,
-     main="NEONCanopy Height Model w Breaks\nHarvard Forest")
+     main="NEON Canopy Height Model \nHarvard Forest")
```

Comment:
This is in the challenge code section.  Changed it so the figure titles were different from where copied and pasted from.  It isn't a main part of this figure.  Sure breaks were specified but that doesn't matter to the story the plot is telling in the challenge.